crates/autopilot/src/infra/blockchain/mod.rs

```diff
@@ -106,7 +109,7 @@ impl Ethereum {
         let (transaction, receipt, traces) = tokio::try_join!(
             self.web3.eth().transaction(hash.0.into()),
             self.web3.eth().transaction_receipt(hash.0),
-            self.web3.debug().transaction(hash.0),
+            self.debug_api_web3.debug().transaction(hash.0),
```

Comment:
nit: I think it makes sense to either add a comment here or perhaps even create a new function on `Ethereum` specifically for fetching traces (+ a comment explaining why it has to use the unbuffered version).